pandas/core/frame.py

```diff
@@ -10460,27 +10461,41 @@ def quantile(
         Name: 0.5, dtype: object
         """
         validate_percentile(q)
+        axis = self._get_axis_number(axis)
 
         if not is_list_like(q):
             # BlockManager.quantile expects listlike, so we wrap and unwrap here
-            res = self.quantile(
+            res_df = self.quantile(
                 [q], axis=axis, numeric_only=numeric_only, interpolation=interpolation
             )
-            return res.iloc[0]
+            res = res_df.iloc[0]
+            if axis == 1 and len(self) == 0:
```

Comment:
can we do this in quantile itself?

Reply:
no, these cases are only reached when we dont have any blocks to operate on.

Reply:
hmm really ought to push this down to eliminate this type of logic in the frame itself, ok for now.